src/Generic.Repository/Models/PageAggregation/PageAttrAbstract.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Generic.Repository.Cache;
+using Generic.Repository.Enums;
+using Generic.Repository.Extension.Filter;
+using Generic.Repository.Models.PageAggregation.PageConfig;
+using Generic.Repository.ThrowError;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Generic.Repository.Models.PageAggregation
+{
+    public abstract class PageAttrAbstract<TIn, TOut> : IPage<TOut>
+        where TIn : class
+        where TOut : class
+    {
+        public PageAttrAbstract(IQueryable<TIn> listEntities, IPageConfig config, ICacheRepository cache)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change the visibility of this constructor to 'protected'.](https://app.codacy.com/manual/guilhermecaixeta/Generic.Service.DotNetCore2.2/pullRequest?prid=4961311)